core/corehttp/gateway_handler.go

```diff
@@ -133,6 +133,7 @@ func (i *gatewayHandler) getOrHeadHandler(w http.ResponseWriter, r *http.Request
 	}
 
 	w.Header().Set("X-IPFS-Path", urlPath)
+	w.Header().Set("Access-Control-Allow-Origin", "*")
```

Comment:
Non-browser clients don't read CORS headers, so setting it to anything other than `Access-Control-Allow-Origin: *` would be treating them differently. So unless there's a use-case for treating them differently, we should be able to always set it like the line we're referencing does.